tests/zenko_e2e/backbeat/tests/crr-pause-resume/awsBackend.js

```diff
@@ -0,0 +1,116 @@
+const assert = require('assert');
+const crypto = require('crypto');
+const  { series, parallel, timesSeries, each } = require('async');
+
+const { scalityS3Client, awsS3Client } = require('../../s3SDK');
+const ReplicationUtility = require('../../ReplicationUtility');
+const BackbeatAPIUtility = require('../../BackbeatAPIUtility');
+
+const scalityUtils = new ReplicationUtility(scalityS3Client);
+const awsUtils = new ReplicationUtility(awsS3Client);
+const backbeatAPIUtils = new BackbeatAPIUtility();
+const srcBucket = `source-bucket-${Date.now()}`;
+const destBucket = process.env.AWS_S3_BACKBEAT_BUCKET_NAME;
+const destLocation = process.env.AWS_S3_BACKEND_DESTINATION_LOCATION;
+const hex = crypto.createHash('md5')
+    .update(Math.random().toString())
+    .digest('hex');
+const keyPrefix = `${srcBucket}/${hex}`;
+const key = `${keyPrefix}/object-to-replicate-${Date.now()}`;
+const key2 = `${key}-2`;
+// eslint-disable-next-line
+const REPLICATION_TIMEOUT = 300000;
+
+describe('Replication Pause-Resume with AWS backend', function() {
+    this.timeout(REPLICATION_TIMEOUT);
+    let roleArn = 'arn:aws:iam::root:role/s3-replication-role';
+
+    beforeEach(done => series([
+        next => scalityUtils.createVersionedBucket(srcBucket, next),
+        next => scalityUtils.putBucketReplicationMultipleBackend(srcBucket,
+            destBucket, roleArn, destLocation, next),
+    ], done));
+
+    afterEach(done => series([
+        next => scalityUtils.deleteVersionedBucket(srcBucket, next),
+        next => awsUtils.deleteAllVersions(destBucket,
+            `${srcBucket}/${keyPrefix}`, next),
+    ], done));
+
+    after(done => backbeatAPIUtils.resumeReplication(null, null, null, done));
+
+    it('should pause and resume replication', done => series([
+        next => scalityUtils.putObject(srcBucket, key, Buffer.alloc(1), next),
+        next => scalityUtils.waitUntilReplicated(srcBucket, key,
+            undefined, next),
+        next => backbeatAPIUtils.pauseReplication(null, next),
+        next => backbeatAPIUtils.getReplicationStatus(null, (err, data) => {
+            assert.ifError(err);
+            assert(data[destLocation]);
+            assert.strictEqual(data[destLocation], 'disabled');
+            return next();
+        }),
+        next => scalityUtils.putObject(srcBucket, key2, Buffer.alloc(1),
+            next),
+        next => awsUtils.assertNoObject(destBucket, key2, next),
```

Comment:
Do we wait some amount of time between the PUT on source and this call? I think it would improve the chances to fail if replication is not paused correctly. Probably 15s or less is enough as we don't want to increase the test run time too much.

Reply:
mm, if we want to make sure replication is paused correctly, the pause would probably go somewhere between `backbeatAPIUtils.pauseReplication` and  `scalityUtils.putObject`?